iroh-util/src/lock.rs

```diff
@@ -0,0 +1,112 @@
+use anyhow::Result;
+use file_lock::{FileLock, FileOptions};
```

Comment:
`file_lock` is basically a `C` file, any chance we can find a library that is actually written in rust 😓

Reply:
maybe: https://github.com/brunoczim/fslock it looks a little more cross platform as well